consent_manager/api.rst

```diff
@@ -0,0 +1,250 @@
+.. highlight:: js
+.. default-domain:: js
+
+JavaScript API
+==============
+
+Introduction
+------------------------
+
+Consent Manager provide JavaScript API that allows the user to:
+
+    * get consent types
+    * get new consent types
+    * get consent settings
+    * set consent settings
+    * send data subject request
+
+JavaScript API is implemented by providing global JavaScript objects queue responsible for executing command:
+
+.. function:: ppms.cm.api(command, ...args)
+.. function:: dataLayer.push(event: command, ...args)
```

Comment:
dataLayer.push is not deprecated. We use dataLayer.push internaly even we call ppms.cm.api, becase we want avoid BC. We can add information that dataLayer.push is not recommended, but it will be fully supported in future. 

We also use dataLayer.push({event: string, ...args}) for others internal Tag Manager events like pageView and this is also way to provide API for old snippets.